smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/WaiterGenerator.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering
+
+import software.amazon.smithy.jmespath.ExpressionVisitor
+import software.amazon.smithy.jmespath.JmespathExpression
+import software.amazon.smithy.jmespath.RuntimeType
+import software.amazon.smithy.jmespath.ast.*
+import software.amazon.smithy.kotlin.codegen.core.RenderingContext
+import software.amazon.smithy.kotlin.codegen.core.defaultName
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.waiters.Matcher
+import software.amazon.smithy.waiters.PathMatcher
+import software.amazon.smithy.waiters.WaitableTrait
+import software.amazon.smithy.waiters.Waiter
+
+class WaiterGenerator(
```

Comment:
**Comment**: This shouldn't be here and I'll pull it out shortly...ignore it for now!